index.js

```diff
@@ -129,6 +134,25 @@ module.exports.init = function () {
       unix_sockets.push(address.address);
       fs.chmodSync(address.address, '664');
     }
+  }).on('exit', function (worker, code, signal) {
+    const pid = worker.process.pid;
+
+    if (worker._worker_terminated) {
+      // this is used here to distinguish expected/unexpected worker deaths.
+      if (code === 0) {
+        debug('PID/%s: terminated worker has exited', pid);
+      } else {
+        debug('PID/%s: terminated worker has crashed: %o', pid, { code, signal });
+      }
+    } else {
+      const uptime = Date.now() - worker._worker_started;
+      const restartDelay = Math.max(0, RESTART_DELAY - uptime);
```

Comment:
I like it, however, I misunderstood what `RESTART_DELAY` was by the name + readme, it is basically the minimum uptime the worker needs to live before restarting on unexpected exit, if it already lived longer it gets ignored.

Reply:
Renamed to `WORKER_THROTTLE` and clarified in README.